go.mod

```diff
@@ -1,10 +1,10 @@
 module github.com/hashicorp/terraform-plugin-framework
 
-go 1.20
+go 1.21
 
 require (
 	github.com/google/go-cmp v0.6.0
-	github.com/hashicorp/terraform-plugin-go v0.21.0
+	github.com/hashicorp/terraform-plugin-go v0.22.0
```

Comment:
Running `go get github.com/hashicorp/terraform-plugin-go` results in the addition of the following entry to `go.mod`:

```
toolchain go1.21.6
```

Which would result in an analogous [CI failure](https://github.com/hashicorp/terraform-plugin-mux/actions/runs/8017456549/job/21901361119#step:4:7) to that seen on _terraform-plugin-mux_

Given that our CI runs both Go `1.20` and `1.21`, I removed this directive, but perhaps this should be handled differently?

Reply:
Oh interesting. It sounds like what you've done is basically the solution: https://go.dev/doc/toolchain

> The same requirement applies in reverse when downgrading: if the go.mod starts at go 1.22.1 and toolchain go1.24rc1, then go get toolchain@go1.22.9 will update only the toolchain line, but go get toolchain@go1.21.3 will downgrade the go line to go 1.21.3 as well. The effect will be to leave just go 1.21.3 with no toolchain line.
>
> The special form toolchain@none means to remove any toolchain line, as in go get toolchain@none or go get go@1.25.0 toolchain@none.

Sounds like we could also potentially just run a `go get toolchain@none` before we run our CI, but you've essentially done that..... Will a `go mod tidy` bring it back?

Reply:
Running `go mod tidy` will not re-instate the toolchain directive.